src/dialogs/more-info/controls/more-info-weather.ts

```diff
@@ -55,7 +56,14 @@ class MoreInfoWeather extends LitElement {
       return nothing;
     }
 
-    const hourly = isForecastHourly(this.stateObj.attributes.forecast);
+    const forecast = getForecast(
+      this.stateObj.attributes.forecast,
+      this.stateObj.attributes.forecast_daily,
+      this.stateObj.attributes.forecast_hourly,
+      this.stateObj.attributes.forecast_twice_daily
+    );
+
+    const hourly = isForecastHourly(forecast);
```

Comment:
We can know what type of forecast `getForecast` returned, only if it is legacy we would have to use this function to know if it is an hourly forecast.

Reply:
Also in [ui-weather-forecast-card.ts](https://github.com/home-assistant/frontend/pull/15028/files#diff-4f7195942cc7cdfbd5ad6149411b2d6fb641035063b8711026e641d7f1a2a606R188)